sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/AnomalyDetectionConfigurationTests.cs

```diff
@@ -0,0 +1,167 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using Azure.AI.MetricsAdvisor.Administration;
+using Azure.AI.MetricsAdvisor.Models;
+using Azure.Core.TestFramework;
+using NUnit.Framework;
+
+namespace Azure.AI.MetricsAdvisor.Tests
+{
+    public class AnomalyDetectionConfigurationTests : ClientTestBase
+    {
+        public AnomalyDetectionConfigurationTests(bool isAsync) : base(isAsync)
+        {
+        }
+
+        private string FakeGuid => "00000000-0000-0000-0000-000000000000";
+
+        [Test]
+        public void CreateDetectionConfigurationValidatesArguments()
+        {
+            MetricsAdvisorAdministrationClient adminClient = GetMetricsAdvisorAdministrationClient();
+
+            Assert.That(() => adminClient.CreateDetectionConfigurationAsync(null), Throws.InstanceOf<ArgumentNullException>());
+
+            Assert.That(() => adminClient.CreateDetectionConfiguration(null), Throws.InstanceOf<ArgumentNullException>());
+        }
+
+        [Test]
+        public void CreateDetectionConfigurationRespectsTheCancellationToken()
+        {
+            MetricsAdvisorAdministrationClient adminClient = GetMetricsAdvisorAdministrationClient();
+
+            var config = new AnomalyDetectionConfiguration(FakeGuid, "configName", new ());
```

Comment:
just curious, is there any test that will verify the required input for `AnomalyDetectionConfiguration`?

Reply:
There isn't. However, we'll change some of our model's designs to have parameterless constructors (including `AnomalyDetectionConfiguration`), so these tests would be discarded. This is feedback we got during the archboard review (https://github.com/Azure/azure-sdk-for-net/issues/16321).

Reply:
We'll still have models that may need validation after that, though. That's a good point, so just opened https://github.com/Azure/azure-sdk-for-net/issues/17650 to track this. We can work on that after the design change.